### ISSUE TITLE
Can initialize function have a proper type?

### ISSUE THREAD

**Issue by linonetwo** [3mcd/javelin#111]
Currently following code will have an any type for the `position`

```js
import { createComponentType, string, number } from '@javelin/ecs';

export const VisibleItem = createComponentType({
  name: 'VisibleItem',
  type: 1,
  schema: {
    x: number,
    y: number,
    texture: string,
  },
  initialize: (position, x = 0, y = 0) => {
    position.x = x;
    position.y = y;
  },
});
```

![Screen Shot 2020-09-06 at 10 52 44 PM](https://user-images.githubusercontent.com/3746270/92328498-d1446280-f093-11ea-93e2-9c38c1876083.png)

Is it possible to properly type `initialize?: I;` using information from the `schema: S;`?

**Comment by 3mcd** [comment]
Yeah, that's how it should work! What version of TypeScript are you using?

**Comment by 3mcd** [comment]
This is fixed in v0.19.2, which will be released shortly!
